python-lib/dku_data_processing/format.py

```diff
@@ -71,42 +126,8 @@ def fetch_geo_data(dss_dataset: Dataset, geopoint, detail, tooltips, filters):
 
     df = dss_dataset.get_dataframe(limit=1000000, columns=column_to_retrieves)
 
-    if (filters is not None) and (filters != []):
-        df = filter_dataframe(df, filters)
-
-    if (detail is None) or (df.dtypes[detail] not in ['int64', 'float64']):
-        df['detail_'] = 1
-    else:
-        df['detail_'] = normalize(df[detail].values)
-
-    if (detail is not None) and (df.dtypes[detail] not in ['int64', 'float64']):
-        raise ValueError("The detail column should be a int or float type")
-
-    def convert(x):
-        if x is not None:
-            return float(x)
-        else:
-            return None
+    geodata_object = extract_df(df, detail, filters, geopoint, tooltips)
```

Comment:
Extract backend function to process dataframe instead of DSS dataset (allow python unit tests)